File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    heap = scoville[:]
    heapq.heapify(heap)
    result = 0
    while heap:
        if heap[0] >= K:
            return result
        
        first = heapq.heappop(heap)
        second = 0
        if heap:
            second = heapq.heappop(heap)
        else:
            return -1
        result += 1
        heapq.heappush(heap, first + second * 2)

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;
import java.util.Arrays.*;

class Solution {
    public int solution(int[] scoville, int K) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        int result = 0;
        for (int i = 0; i < scoville.length; i++) {
            pq.offer(scoville[i]);
        }
        
        while (true) {
            if (pq.peek() >= K) {
                return result;
            }
            if (pq.size() <= 1)
                break;
            int a = pq.poll();
            
            int b = pq.poll();
            
            pq.offer(mix(a, b));
            result++;
        }
        return -1;
    }
    
    static int mix(int a, int b) {
        return a + b * 2;
    }
}